src/main/kotlin/by/architecture_map/belarus/service/impl/ArchitecturalStyleServiceImpl.kt

```diff
@@ -0,0 +1,76 @@
+package by.architecture_map.belarus.service.impl
+
+import by.architecture_map.belarus.entity.ArchitecturalStyle
+import by.architecture_map.belarus.repository.ArchitecturalStyleRepository
+import by.architecture_map.belarus.repository.ImageRepository
+import by.architecture_map.belarus.service.ArchitecturalStyleService
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+
+@Service
+class ArchitecturalStyleServiceImpl(
+        private val architecturalStyleRepository: ArchitecturalStyleRepository,
+        private val imageRepository: ImageRepository
+) : ArchitecturalStyleService {
+
+    override fun create(architecturalStyle: ArchitecturalStyle): ArchitecturalStyle {
+        return architecturalStyleRepository.save(architecturalStyle)
+    }
+
+    override fun findAll(): MutableList<ArchitecturalStyle> {
+        return architecturalStyleRepository.findAll()
+    }
+
+    override fun findById(id: Int): ArchitecturalStyle? {
+        return architecturalStyleRepository.findByIdOrNull(id)
+    }
+
+    override fun updateById(id: Int, updatedArchitecturalStyle: ArchitecturalStyle): ArchitecturalStyle? {
```

Comment:
the method could be implemented like that, without any additional variables:
```
    override fun updateById(id: Int, updatedArchitecturalStyle: ArchitecturalStyle): ArchitecturalStyle =
        architecturalStyleRepository.findById(id)
            ?.apply {
                name = updatedArchitecturalStyle.name
                attributes = updatedArchitecturalStyle.attributes
                description = updatedArchitecturalStyle.description
                shortDescription = updatedArchitecturalStyle.shortDescription
                demonstrativeImage = updatedArchitecturalStyle.demonstrativeImage
                yearsActive = updatedArchitecturalStyle.yearsActive
            }
            ?.let { architecturalStyleRepository.save(it) }
            ?: throw EntityNotFountException()
```

for better understanding of scope functions, please check the documentation:
https://kotlinlang.org/docs/scope-functions.html

Reply:
please update other methods